pkgs/development/python-modules/pyscf/default.nix

```diff
@@ -28,12 +31,57 @@ buildPythonPackage rec {
     numpy
     scipy
     h5py
+    cppe
   ];
 
+  checkInputs = [ nose nose-exclude ];
+
   PYSCF_INC_DIR="${libcint}:${libxc}:${xcfun}";
 
-  doCheck = false;
   pythonImportsCheck = [ "pyscf" ];
+  preCheck = ''
+    # Set config used by tests to ensure reproducibility
+    echo 'pbc_tools_pbc_fft_engine = "NUMPY"' > pyscf/pyscf_config.py
+    export OMP_NUM_THREADS=$NIX_BUILD_CORES
+    export PYSCF_CONFIG_FILE=$(pwd)/pyscf/pyscf_config.py
+  '';
+  checkPhase = ''
+    runHook preCheck
+
+    nosetests pyscf/ -v \
+        --exclude-dir=examples --exclude-dir=pyscf/pbc/grad \
```

Comment:
(optional): these excludes can be passed as environment variables, e.g. ``NOSE_EXCLUDE=["test_bz", "h2o_vdz", ...];``

This allows commenting the individual tests better (i.e. this is actually broken, this is slow, etc.)

Env var names/options correlation can be found here: https://nose.readthedocs.io/en/latest/usage.html#options

Reply:
Yes, I've tried this version as in your old version of the derivation. I am not sure why, but it didn't have any effect, whether I put them only in the derivation directly, as your old code did, or export them in the `preCheck` explicitly.

Reply:
Did you have the ``nose-exclude`` plugin included on the path (``checkInputs``)? If so, weird. I don't know why that would be.

Reply:
Yes, was available :open_mouth: I am also a little bit lost there, as I can't find the difference to you version. Anyway, I'm approaching an end with getting the test suite to run.